File: 3634_Minimum_Removals_to_Balance_Array/solution.py
class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        nums.sort()
        max_len, low = 1, 0
        for high in range(len(nums)):
            while low < high and nums[high] > nums[low] * k:
                low += 1
            max_len = max(max_len, high - low + 1)
        return len(nums) - max_len

File: 3634_Minimum_Removals_to_Balance_Array/solution.c
int comp(const void* a, const void* b) { return (*(int*)a - *(int*)b); }
int minRemoval(int* nums, int numsSize, int k) {
    qsort(nums, numsSize, sizeof(int), comp);
    int max_len = 1, low = 0;
    for (int high = 0; high < numsSize; high++) {
        while (low < high && nums[high] > (long long)nums[low] * k) low++;
        if (high - low + 1 > max_len) max_len = high - low + 1;
    }
    return numsSize - max_len;
}